controllers/argocd/util.go

```diff
@@ -1083,9 +1103,11 @@ func (r *ReconcileArgoCD) setResourceWatches(bldr *builder.Builder, clusterResou
 
 	bldr.Watches(&v1.ClusterRole{}, clusterResourceHandler)
 
-	bldr.Watches(&corev1.ConfigMap{ObjectMeta: metav1.ObjectMeta{
-		Name: common.ArgoCDAppSetGitlabSCMTLSCertsConfigMapName,
-	}}, appSetGitlabSCMTLSConfigMapHandler)
+	if applicationSetSCMTLSConfigMapName != "" {
+		bldr.Watches(&corev1.ConfigMap{ObjectMeta: metav1.ObjectMeta{
+			Name: applicationSetSCMTLSConfigMapName,
+		}}, appSetGitlabSCMTLSConfigMapHandler)
+	}
```

Comment:
I think the watch will be setup only for initial value of `applicationSetSCMTLSConfigMapName` when the operator manager starts. Updates to `applicationSetSCMTLSConfigMapName` won't reset the watch with new value as the watches are set only at the start of the manager. 

What is the purpose of this watch? Is it to reflect changes in configmap to the appset-controller deployment or to point the appset-controller deployment to the new cm name whenever it changes in `ArgoCD` CR? 
IMO, reflecting configmap content changes should be handled in upstream Argo CD. I think it already does for `argocd-server` deployments where changes to `argocd-cm` configmap are automatically reflected without a deployment restart. However, I don't know if it handles tls content. 
The name change in ArgoCD CR doesn't necessarily require a separate watch, existing ArgoCD CR watch should be able to handle this update in configmap name change.

Reply:
As discussed, not updating the watch logic and updated the doc to use a specific config map name so that the watch can be specified.